go-controller/pkg/clustermanager/node/node_allocator.go

```diff
@@ -230,6 +237,13 @@ func (na *NodeAllocator) HandleAddUpdateNodeEvent(node *corev1.Node) error {
 				}
 				return fmt.Errorf("failed to set hybrid overlay annotations for node %s: %v", node.Name, err)
 			}
+		} else if na.hasHybridOverlayAllocationUnmanaged() {
+			// this is a hybrid overlay node but not managed by the hybrid overlay subnet allocator
+			// Hybrid overlay is only available for IPv4 for now, so we only need to check for IPv4 subnets
+			nodeSubnets := na.clusterSubnetAllocator.ListAllIPv4Networks()
```

Comment:
you are already initializing nodeSubnets in `markAllocatedNetworksForUnmanagedHONode` so it wouldn't be needed here